packages/datatrak-web/src/views/AccountSettingsPage/PersonalDetailsSection.tsx

```diff
@@ -4,10 +4,168 @@
  */
 
 import React from 'react';
+import styled from 'styled-components';
+import { useForm, Controller, SubmitHandler } from 'react-hook-form';
 import { AccountSettingsSection } from './AccountSettingsSection';
+import { Button } from '../../components';
+import { successToast } from '../../utils';
+import { TextField } from '@tupaia/ui-components';
+import { useCurrentUser } from '../../api';
+import { useEditUser } from '../../api/mutations';
+import { UserAccountDetails } from '../../types';
+
+type PersonalDetailsFormFields = Pick<
+  UserAccountDetails,
+  'firstName' | 'lastName' | 'employer' | 'position' | 'mobileNumber'
+>;
+
+const ButtonWrapper = styled.div`
+  grid-column: -2;
+`;
+
+const StyledTextField = styled(TextField)`
+  margin: 0; // Use gap on parent to control spacing
+`;
+
+const PersonalDetailsForm = styled.form`
+  display: grid;
+  gap: 1.56rem 1.25rem;
+  max-width: 44.25rem;
+  width: 100%;
+  ${({ theme }) => theme.breakpoints.up('sm')} {
+    grid-template-columns: repeat(2, 1fr);
+  }
+
+  .MuiFormLabel-root {
+    color: ${props => props.theme.palette.text.primary};
+    font-weight: ${({ theme }) => theme.typography.fontWeightMedium};
+  }
+`;
 
 export const PersonalDetailsSection = () => {
+  const { mutate: updateUser } = useEditUser(() => successToast('Personal details updated'));
+  const user = useCurrentUser();
+
+  const {
+    control,
+    formState: { isDirty, dirtyFields },
+    handleSubmit,
+    register,
+  } = useForm<PersonalDetailsFormFields>({
+    defaultValues: {
+      firstName: user.firstName ?? '',
+      lastName: user.lastName ?? '',
+      mobileNumber: user.mobileNumber ?? '',
+      employer: user.employer ?? '',
+      position: user.position ?? '',
+    },
+  });
+
+  function onSubmit(userDetails): SubmitHandler<PersonalDetailsFormFields> {
+    const updates = Object.keys(dirtyFields)
+      .filter(field => dirtyFields[field])
+      .reduce(
+        (updatedFields, field) => ({ ...updatedFields, [field]: userDetails[field] }),
+        {} as UserAccountDetails,
+      );
```

Comment:
It was a toss-up between the implementation here versus:

```ts
const updates2 = Object.fromEntries(
  Object.entries(userDetails).filter(([field]) => dirtyFields[field]),
);
```

I’m not sure which is more performant and/or readable. There’s a similar use of `.reduce()` in `useEditUser()`, but I’m open to opinions!

Reply:
I think it's very similar so no strong opinion on this. If I had to pick one I would go for the latter as I find reducers a tad harder to read